File: sdkwork-backend-sdk-typescript/src/api/rbac-user-role.ts
import { backendApiPath } from './paths';
import type { HttpClient } from '../http/client';
import type { QueryParams } from '../types/common';
import type { PlusApiResultBoolean, PlusApiResultListPlusUserRoleVO, PlusApiResultPagePlusUserRoleVO, PlusApiResultPlusUserRoleVO, PlusUserRoleForm, QueryListForm } from '../types';


export class RbacUserRoleApi {
  private client: HttpClient;
  
  constructor(client: HttpClient) { 
    this.client = client; 
  }

/** Update user-role relationship */
  async update(body: PlusUserRoleForm): Promise<PlusApiResultPlusUserRoleVO> {
    return this.client.put<PlusApiResultPlusUserRoleVO>(backendApiPath(`/rbac/user/role`), body);
  }

/** Create user-role relationship */
  async create(body: PlusUserRoleForm): Promise<PlusApiResultPlusUserRoleVO> {
    return this.client.post<PlusApiResultPlusUserRoleVO>(backendApiPath(`/rbac/user/role`), body);
  }

/** Get user-role relationships by page */
  async listByPage(body?: QueryListForm, params?: QueryParams): Promise<PlusApiResultPagePlusUserRoleVO> {
    return this.client.post<PlusApiResultPagePlusUserRoleVO>(backendApiPath(`/rbac/user/role/list`), body, params);
  }

/** Get all user-role relationships */
  async listAllEntities(body?: QueryListForm): Promise<PlusApiResultListPlusUserRoleVO> {
    return this.client.post<PlusApiResultListPlusUserRoleVO>(backendApiPath(`/rbac/user/role/list/all`), body);
  }

/** Get user-role relationship by ID */
  async getById(id: string | number): Promise<PlusApiResultPlusUserRoleVO> {
    return this.client.get<PlusApiResultPlusUserRoleVO>(backendApiPath(`/rbac/user/role/${id}`));
  }

/** Delete user-role relationship */
  async delete(id: string | number): Promise<PlusApiResultBoolean> {
    return this.client.delete<PlusApiResultBoolean>(backendApiPath(`/rbac/user/role/${id}`));
  }
}

export function createRbacUserRoleApi(client: HttpClient): RbacUserRoleApi {
  return new RbacUserRoleApi(client);
}

File: sdkwork-backend-sdk-go/api/rbac_user_role.go
package api

import (
    "fmt"
    sdktypes "github.com/sdkwork/backend-sdk/types"
    sdkhttp "github.com/sdkwork/backend-sdk/http"
)

type RbacUserRoleApi struct {
    client *sdkhttp.Client
}

func NewRbacUserRoleApi(client *sdkhttp.Client) *RbacUserRoleApi {
    return &RbacUserRoleApi{client: client}
}

// Update user-role relationship
func (a *RbacUserRoleApi) Update(body sdktypes.PlusUserRoleForm) (sdktypes.PlusApiResultPlusUserRoleVO, error) {
    raw, err := a.client.Put(BackendApiPath("/rbac/user/role"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultPlusUserRoleVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPlusUserRoleVO](raw)
}

// Create user-role relationship
func (a *RbacUserRoleApi) Create(body sdktypes.PlusUserRoleForm) (sdktypes.PlusApiResultPlusUserRoleVO, error) {
    raw, err := a.client.Post(BackendApiPath("/rbac/user/role"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultPlusUserRoleVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPlusUserRoleVO](raw)
}

// Get user-role relationships by page
func (a *RbacUserRoleApi) ListByPage(body *sdktypes.QueryListForm, query map[string]interface{}) (sdktypes.PlusApiResultPagePlusUserRoleVO, error) {
    raw, err := a.client.Post(BackendApiPath("/rbac/user/role/list"), body, query, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultPagePlusUserRoleVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPagePlusUserRoleVO](raw)
}

// Get all user-role relationships
func (a *RbacUserRoleApi) ListAllEntities(body *sdktypes.QueryListForm) (sdktypes.PlusApiResultListPlusUserRoleVO, error) {
    raw, err := a.client.Post(BackendApiPath("/rbac/user/role/list/all"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultListPlusUserRoleVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultListPlusUserRoleVO](raw)
}

// Get user-role relationship by ID
func (a *RbacUserRoleApi) GetById(id string) (sdktypes.PlusApiResultPlusUserRoleVO, error) {
    raw, err := a.client.Get(BackendApiPath(fmt.Sprintf("/rbac/user/role/%s", id)), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultPlusUserRoleVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPlusUserRoleVO](raw)
}

// Delete user-role relationship
func (a *RbacUserRoleApi) Delete(id string) (sdktypes.PlusApiResultBoolean, error) {
    raw, err := a.client.Delete(BackendApiPath(fmt.Sprintf("/rbac/user/role/%s", id)), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultBoolean
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultBoolean](raw)
}
